core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -916,6 +915,16 @@ public AggImplementor get(final SqlAggFunction aggregation,
     return aggSupplier.get();
   }
 
+  public MatchImplementor get(final SqlMatchFunction function) {
+    final Supplier<? extends MatchImplementor> supplier = mrMap.get(function);
+    if (supplier != null) {
+      return supplier.get();
+    } else {
+      // TODO throw an exception here?
+      return null;
```

Comment:
Personally i don't like the `TODOs` in the code, we better log a JIRA issue for it if we are planning to resolve it.